lib/algos/list/skip.js

```diff
@@ -0,0 +1,88 @@
+const assert = require('assert');
+
+const { FILTER_END, FILTER_SKIP, SKIP_NONE } = require('./tools');
+
+
+const MAX_STREAK_LENGTH = 100;
+
+/**
+ * Handle the filtering and the skip mechanism of a listing result.
+ */
+class Skip {
+    /**
+     * @param {Object} params           - skip parameters
+     * @param {Object} params.extension - delimiter extension used (required)
+     * @param {String} params.gte       - current range gte (greater than or
+     *                                    equal) used by the client code
+     */
+    constructor(params) {
+        assert(params.extension);
+
+        this.extension = params.extension;
+        this.gteParams = params.gte;
+
+        this.listingEndCb = null;
+        this.skipRangeCb = null;
+
+        /* Used to count consecutive FILTER_SKIP returned by the extension
+         * filter method. Once this counter reaches MAX_STREAK_LENGTH, the
+         * filter function tries to skip unwanted values by defining a new
+         * range. */
+        this.streakLength = 0;
+    }
+
+    setListingEndCb(cb) {
+        this.listingEndCb = cb;
```

Comment:
`listingEndCb` and `skipRangeCb` contracts should be documented explicitly, since they are essential to the behavior of the listing extension.